boards/riscv/longan_nano/longan_nano_defconfig

```diff
@@ -10,6 +10,8 @@ CONFIG_BOARD_LONGAN_NANO=y
 
 CONFIG_GD32_HXTAL_8MHZ=y
 
+CONFIG_FLASH_BASE_ADDRESS=0x08000000
```

Comment:
Normally it auto detected through `dt_chosen_reg_addr_hex`. I add PR #54967 to fix this issue.

Reply:
do i understand correctly that i should cherry-pick 95da6b56da85ec23443f6d28662f7f4159210846, and divide this MR into separate commits (adding the flash-config to longan_nano, and adding the dfu settings)?

Reply:
This file may no need to change because the CONFIG_FLASH_BASE be set correctly by `arch/Kconfig` settings that enabled by this PR. (Please check generated .config file.)

Reply:
This file may no need to change because the CONFIG_FLASH_BASE be set correctly by arch/Kconfig settings that enabled by this PR. (Please check generated .config file.)

Reply:
Off topic:
Reviewers reviews a lot of PRs, and they press this button to indicate that their issue has been resolved.
It is better not to press this button yourself.
(People forget easy what say even yesterday. The `Resolve conversation` button works as reviewer's reminder.)

Reply:
Hi @ottojo , No need to cherry-pick, you already make very well works. It just my suggestion.